python/ray/experimental/sgd/sgd_worker.py

```diff
@@ -113,7 +115,12 @@ def __init__(self,
             manager_socket = (
                 ray.worker.global_worker.plasma_client.manager_socket_name)
             if not plasma.tf_plasma_op:
+                lock_path = os.path.join(pyarrow.__path__[0], "tensorflow",
+                                         "compile_op.lock")
+                lock = SoftFileLock(lock_path)
+                lock.acquire()
                 plasma.build_plasma_tensorflow_op()
+                lock.release()
```

Comment:
Instead of calling `plasma.build_plasma_tensorflow_op()` here, maybe we should provide a wrapped version that includes the filelock and call that here.